pkg/agent/openflow/client.go

```diff
@@ -812,6 +799,65 @@ func (c *client) Initialize(roundInfo types.RoundInfo, nodeConfig *config.NodeCo
 	return connCh, c.initialize()
 }
 
+func (c *client) initializeFeatures(isIPv4Enabled bool) map[pipelineProtocol][]*featureTemplate {
+	var features []feature
+	podConnectivityIP := newFeaturePodConnectivity(c.cookieAllocator,
+		c.ipProtocols,
+		c.nodeConfig,
+		c.networkConfig,
+		c.connectUplinkToBridge)
+	c.featurePodConnectivity = podConnectivityIP.(*featurePodConnectivity)
```

Comment:
as far as I can tell, there is no reason to return the interface (`feature`) in the `newFeaturePodConnectivity` as opposed to the actual pointer (`* featurePodConnectivity`). It's pretty unusual to do so. Returning the actual pointer would remove the need to have a type assertion.

Reply:
Your are right, there is no need to return the interface. Just the actual pointer is ok.